tests/testsuite/profile_trim_paths.rs

```diff
@@ -511,3 +511,69 @@ fn object_works() {
     assert!(memchr::memmem::find(&stdout, registry_src_bytes).is_none());
     assert!(memchr::memmem::find(&stdout, pkg_root).is_none());
 }
+
+// TODO: might want to move to test/testsuite/build_script.rs once stabilized.
+#[cargo_test(nightly, reason = "-Zremap-path-scope is unstable")]
+fn custom_build_env_var_trim_paths() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+           "#,
+        )
+        .file("src/lib.rs", "")
+        .file("build.rs", "")
+        .build();
+
+    let test_cases = [
+        ("none", "false"),
+        ("none", "\"none\""),
+        ("none", "[]"),
+        ("all", "true"),
+        ("all", "\"all\""),
+        ("diagnostics", "\"diagnostics\""),
+        ("macro", "\"macro\""),
+        ("object", "\"object\""),
+        (
+            "diagnostics,macro,object",
+            r#"["diagnostics", "macro", "object"]"#,
+        ),
+    ];
+
+    for (expected, opts) in test_cases {
```

Comment:
nit: having `expected` first threw me off and made it harder to read.

Reply:
Swapped. Thanks!